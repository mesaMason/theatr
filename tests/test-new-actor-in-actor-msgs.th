dolphin(int weight, int age):
    print("dolphin here")
    print(weight)
    print(age)
    actor baby = new babyDolphin(weight*2, age*2)
    receive:
        eat(int food):
            weight = weight + food
            print("dolphin eating, now weighs")
            print(weight)
            babyDolphin.eat(55) | baby
            babyDolphin.die() | baby
        growOld(int time):
            age = age + time
        eatAndGrowOld(int food, int time):
            weight = weight + food
            age = age + time
    drop:
        print("inside drop()")
        
babyDolphin(int weight, int age):
    receive:
        eat(int food):
            weight = weight + food
            print("baby dolphin eating, now weighs")
            print(weight)
    drop:
        print("inside drop()")

func main() -> int:
    int weight = 100
    int age = 3
    actor d = new dolphin(weight, age)
    dolphin.eat(10) | d
    dolphin.die() | d
    return 0
