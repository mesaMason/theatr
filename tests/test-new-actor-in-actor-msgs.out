dolphin here
100
3
dolphin eating, now weighs
110
baby dolphin eating, now weighs
255